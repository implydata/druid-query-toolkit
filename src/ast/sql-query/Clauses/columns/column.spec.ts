/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Column, OrExpression, StringType } from '../../../index';

describe('single column Tests', () => {
  it('column with no brackets to string', () => {
    const val = new Column({
      parens: [],
      ex: new OrExpression({
        basicExpression: new StringType({ chars: 'value', quote: "'", spacing: ['', ''] }),
      }),
      alias: null,
      spacing: [''],
    });
    expect(val.toString()).toMatchSnapshot();
  });
  it('column addparen', () => {
    const val = new Column({
      parens: [],
      ex: new OrExpression({
        basicExpression: new StringType({ chars: 'value', quote: "'", spacing: ['', ''] }),
      }),
      alias: null,
      spacing: [''],
    });
    val.addParen(['(', [null]], [[null], ')']);
    expect(val).toMatchSnapshot();
  });
  it('column addparen to string', () => {
    const val = new Column({
      parens: [],
      ex: new OrExpression({
        basicExpression: new StringType({ chars: 'value', quote: "'", spacing: ['', ''] }),
      }),
      alias: null,
      spacing: [''],
    });
    val.addParen(['(', [null]], [[null], ')']);
    expect(val.toString()).toMatchSnapshot();
  });
  it('column getBasicValue with bracket', () => {
    const val = new Column({
      parens: [],
      ex: new OrExpression({
        basicExpression: new StringType({ chars: 'value', quote: "'", spacing: ['', ''] }),
      }),
      alias: null,
      spacing: [''],
    });
    val.addParen(['(', [null]], [[null], ')']);
    expect(val.getBasicValue()).toMatchSnapshot();
  });
  it('column getAlias', () => {
    const val = new Column({
      parens: [],
      ex: new OrExpression({
        basicExpression: new StringType({ chars: 'value', quote: "'", spacing: ['', ''] }),
      }),
      alias: null,
      spacing: [''],
    });
    val.addParen(['(', [null]], [[null], ')']);
    expect(val.getBasicValue()).toMatchSnapshot();
  });
});
