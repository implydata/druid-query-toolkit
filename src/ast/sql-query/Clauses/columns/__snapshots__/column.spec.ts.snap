// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single column Tests column addparen 1`] = `
Column {
  "alias": null,
  "ex": OrExpression {
    "ex": Array [
      OrPart {
        "ex": AndExpression {
          "ex": Array [
            AndPart {
              "ex": NotExpression {
                "ex": ComparisonExpression {
                  "ex": AdditiveExpression {
                    "ex": Array [
                      MultiplicativeExpression {
                        "ex": Array [
                          String {
                            "chars": "value",
                            "quote": "'",
                            "spacing": Array [
                              "",
                              "",
                            ],
                          },
                        ],
                        "op": null,
                        "parens": Array [],
                        "spacing": Array [
                          "",
                        ],
                      },
                    ],
                    "op": null,
                    "parens": Array [],
                    "spacing": Array [
                      "",
                    ],
                  },
                  "parens": Array [],
                  "rhs": null,
                  "spacing": null,
                },
                "keyword": null,
                "parens": Array [],
                "spacing": null,
              },
              "keyword": "",
              "spacing": Array [
                "",
              ],
            },
          ],
          "parens": Array [],
          "spacing": null,
        },
        "keyword": "",
        "spacing": Array [
          "",
        ],
      },
    ],
    "parens": Array [],
    "spacing": null,
  },
  "parens": Array [
    Object {
      "close": Array [
        Array [
          null,
        ],
        ")",
      ],
      "open": Array [
        "(",
        Array [
          null,
        ],
      ],
    },
  ],
  "spacing": Array [
    "",
  ],
}
`;

exports[`single column Tests column addparen to string 1`] = `"('value')"`;

exports[`single column Tests column getAlias 1`] = `"value"`;

exports[`single column Tests column getBasicValue with bracket 1`] = `"value"`;

exports[`single column Tests column with no brackets to string 1`] = `"'value'"`;
